### PR TITLE
New files with MUI library

### PR DESCRIPTION
**Pull request by zbednarz**
Updated the code with MUI Library

### CODE REVIEW

File: src/styles/MyInput.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { TextField } from "@mui/material";
```
Comment (SMysakowski) [inline src/styles/MyInput.js]:
```suggestion
import TextField from "@mui/material/TextField";
```

https://mui.com/material-ui/guides/minimizing-bundle-size/#development-environment

File: src/styles/MyInput.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+import { styled } from "@mui/system";
+
+const MyInput = styled(TextField)(()=>({
+    background: "white",
+    borderRadius: "15px",
+    fontSize: "60px",     
+}));
```
Comment (SMysakowski) [inline src/styles/MyInput.js]:
```suggestion
const MyInput = styled(TextField)(({theme})=>({
    background: theme.palette.common.white,
    borderRadius: "15px",
    fontSize: "60px",     
}));
```

File: src/styles/MyIcons.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+// import CloseIcon from "@mui/icons-material/Close";
+// import { styled } from "@mui/system";
+
```
Comment (SMysakowski) [inline src/styles/MyIcons.js]: remove unused code

File: src/styles/MyButton.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import { Button } from "@mui/material";
```
Comment (SMysakowski) [inline src/styles/MyButton.js]: import directly

File: src/styles/MyButton.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+import { styled } from "@mui/system";
```
Comment (SMysakowski) [inline src/styles/MyButton.js]: incorrect import

File: src/styles/MyButton.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+const MyButton = styled(Button)(() => ({}));
```
Comment (SMysakowski) [inline src/styles/MyButton.js]: unused

File: src/styles/MyButton.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+export const CompleteButton = styled(MyButton)(() => ({
+  margin: "5px",
+  background: "linear-gradient(45deg, #00c853, #b9f6ca)",
+  color: "white",
+  ":hover": {
+    color: "green",
+    backgroundColor: "white",
+    background: "linear-gradient(45deg, white, white)",
+  },
+}));
```
Comment (SMysakowski) [inline src/styles/MyButton.js]: use colors and spacing from theme

File: src/styles/MyButton.js [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+  color: "secondary",
+  fontSize: "25px",
+  borderRadius: "15px",
+  textTransform: "none",
+  backgroundColor: "white",
+  ":hover": {
+    color: "white",
+    backgroundColor: "darkblue",
+  },
+}));
```
Comment (SMysakowski) [inline src/styles/MyButton.js]: as above

File: src/styles/MyAvatar.js [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+const MyAvatar = styled(Avatar)(()=>({
+               
+}));
+
+export const PositionAvatar = styled(MyAvatar)(()=>({
+    position: "fixed",
+    top: "10px",
+    right: "5px",
+    margin: "5px"
+}));
```
Comment (SMysakowski) [inline src/styles/MyAvatar.js]: use spacing from theme

File: src/styles/MyAvatar.js [hunk 10]
```diff
@@ -0,0 +1,15 @@
+import { Avatar } from "@mui/material";
+import { styled } from "@mui/system";
+
+const MyAvatar = styled(Avatar)(()=>({
+               
+}));
+
```
Comment (SMysakowski) [inline src/styles/MyAvatar.js]: unused

File: src/styles/MyAvatar.js [hunk 11]
```diff
@@ -0,0 +1,15 @@
+import { Avatar } from "@mui/material";
+import { styled } from "@mui/system";
```
Comment (SMysakowski) [inline src/styles/MyAvatar.js]: incorrect import

File: src/styles/MyAvatar.js [hunk 12]
```diff
@@ -0,0 +1,15 @@
+import { Avatar } from "@mui/material";
```
Comment (SMysakowski) [inline src/styles/MyAvatar.js]: import directly. Last comment has explanation

File: src/components/TodoList.js [hunk 13]
```diff
@@ -1,5 +1,6 @@
 import Todo from "./Todo";
 
+
 const TodoList = ({ todos, setTodos, filtered }) => {
   const onDeleter = (todo) => {
```
Comment (SMysakowski) [inline src/components/TodoList.js]: onDelete

File: src/components/Todo.js [hunk 14]
```diff
@@ -1,18 +1,42 @@
...
+
+
 const Todo = ({ text, todo, onComplete, onDeleter }) => {
   return (
+    <ThemeProvider theme={theme}>  
+
     <li className="tasks">
       <li className={`task-item ${todo.completed ? "completed" : ""}`}>
         {text}
       </li>
```
Comment (SMysakowski) [inline src/components/Todo.js]: nested <li>?

File: src/components/Todo.js [hunk 15]
```diff
@@ -1,18 +1,42 @@
...
+import { Stack } from "@mui/system";
+import { ThemeProvider } from "@mui/material/styles";
+import theme from "../styles/styles";
+import {CompleteButton} from "../styles/MyButton";
+import { DeleteButton } from "../styles/MyButton";
+
+
 const Todo = ({ text, todo, onComplete, onDeleter }) => {
   return (
+    <ThemeProvider theme={theme}>
```
Comment (SMysakowski) [inline src/components/Todo.js]: Theme provider should be on the top of app

File: src/components/Todo.js [hunk 16]
```diff
@@ -1,18 +1,42 @@
+
+import CheckIcon from "@mui/icons-material/Check";
+import DeleteIcon from "@mui/icons-material/Delete";
+import { Stack } from "@mui/system";
```
Comment (SMysakowski) [inline src/components/Todo.js]: incorrect import

File: src/components/Modal.js [hunk 17]
```diff
@@ -0,0 +1,57 @@
...
+          <Fade in={open}>
+            <Box sx={style}>
+              <Button onClick={handleClose}>
+                <DehazeIcon
+                  color="orange"
+                  style={{
+                    fontSize: "56",
+                    transform: "translate(-10%, -50%)",
+                  }}
+                />
```
Comment (SMysakowski) [inline src/components/Modal.js]: inline styles

File: src/components/Modal.js [hunk 18]
```diff
@@ -0,0 +1,57 @@
...
+          aria-describedby="modal-modal-description"
+          open={open}
+          closeAfterTransition
+          BackdropComponent={Backdrop}
+          BackdropProps={{
+            timeout: 1000,
+          }}
+        >
+          <Fade in={open}>
+            <Box sx={style}>
```
Comment (SMysakowski) [inline src/components/Modal.js]: you shouldn't use sx. It is slower than styled

File: src/components/Modal.js [hunk 19]
```diff
@@ -0,0 +1,57 @@
...
+
+  return (
+    <ThemeProvider theme={theme}>
+      <div>
+        <Button onClick={handleOpen}>
+          <CloseIcon
+           color="orange"
+            style={{
+              fontSize: "56",
+            }}
```
Comment (SMysakowski) [inline src/components/Modal.js]: inline styles

File: src/components/Modal.js [hunk 20]
```diff
@@ -0,0 +1,57 @@
...
+import { Fade } from "@mui/material";
+import CloseIcon from "@mui/icons-material/Close";
+import DehazeIcon from "@mui/icons-material/Dehaze";
+import { ThemeProvider } from "@mui/material/styles";
+import theme from "../styles/styles";
+import style from "../styles/MyModal";
+
+
+export default function TransitionsModal() {
+  const [open, setOpen] = React.useState(true);
```
Comment (SMysakowski) [inline src/components/Modal.js]: import only useState not whole react

File: src/components/Modal.js [hunk 21]
```diff
@@ -0,0 +1,57 @@
+import * as React from "react";
+import Box from "@mui/material/Box";
+import { Backdrop } from "@mui/material";
+import Button from "@mui/material/Button";
+import Modal from "@mui/material/Modal";
+import { Fade } from "@mui/material";
```
Comment (SMysakowski) [inline src/components/Modal.js]: import directly

File: src/components/Modal.js [hunk 22]
```diff
@@ -0,0 +1,57 @@
...
+import style from "../styles/MyModal";
+
+
+export default function TransitionsModal() {
+  const [open, setOpen] = React.useState(true);
+  const handleOpen = () => setOpen(true);
+  const handleClose = () => setOpen(false);
+
+  return (
+    <ThemeProvider theme={theme}>
```
Comment (SMysakowski) [inline src/components/Modal.js]: move to the top of the app

File: src/components/Form.js [hunk 23]
```diff
@@ -16,26 +27,59 @@ const Form = ({ input, setInput, todos, setTodos, setFilter }) => {
...
 
-      <div className="filter">
-        <select onChange={filterHandler}>
-          <option value="all">All</option>
-          <option value="completed">Done</option>
-          <option value="uncompleted">Pending</option>
-        </select>
-      </div>
-    </form>
+        <Box sx={{ minWidth: 150 }}>
```
Comment (SMysakowski) [inline src/components/Form.js]: move to styled instead sx

File: src/components/Form.js [hunk 24]
```diff
@@ -16,26 +27,59 @@ const Form = ({ input, setInput, todos, setTodos, setFilter }) => {
...
+          
+          <FormControl 
+          variant="standard" 
+          // sx={{}}
```
Comment (SMysakowski) [inline src/components/Form.js]: remove

File: src/components/Form.js [hunk 25]
```diff
@@ -16,26 +27,59 @@ const Form = ({ input, setInput, todos, setTodos, setFilter }) => {
...
-      <button onClick={formSubmit} disabled={!input} type="submit">
-        Add new task
-      </button>
+    <ThemeProvider theme={theme}>
+      <form>
+        <Box
+          sx={{
+            "& .MuiTextField-root": { m: 1, width: "10ch" },
+            "& fieldset": { border: "none" },
+          }}
```
Comment (SMysakowski) [inline src/components/Form.js]: move to styled

File: src/components/Form.js [hunk 26]
```diff
@@ -16,26 +27,59 @@ const Form = ({ input, setInput, todos, setTodos, setFilter }) => {
...
-        type="text"
-        name="name"
-        placeholder="Add Task"
-        value={input}
-        onChange={inputHandler}
-      />
-      <button onClick={formSubmit} disabled={!input} type="submit">
-        Add new task
-      </button>
+    <ThemeProvider theme={theme}>
```
Comment (SMysakowski) [inline src/components/Form.js]: move to top of the app

File: src/Holder.js [hunk 27]
```diff
@@ -0,0 +1,16 @@
...
+    width: "1200px",
+    background: "#fb8c00",
+    borderRadius: "10px",
+    alignItems: "center",
+    justifyContent: "center",
+    transform: "translate(-50%, 0%)",
+    position: "absolute",
+    top: "10%",
+    left: "50%",
+  }));
```
Comment (SMysakowski) [inline src/Holder.js]: to spacing and colors use theme property

File: src/Holder.js [hunk 28]
```diff
@@ -0,0 +1,16 @@
+import { Box } from "@mui/material";
```
Comment (SMysakowski) [inline src/Holder.js]: import directly

File: src/App.js [hunk 29]
```diff
@@ -2,6 +2,20 @@ import { useState, useEffect } from "react";
...
 import TodoList from "./components/TodoList";
+import {
+  Typography,
+  AppBar,
+  Toolbar,
+  Box,
+  Tooltip
+} from "@mui/material";
+
+import { ThemeProvider } from "@mui/material/styles";
```
Comment (SMysakowski) [inline src/App.js:11]: wrap theme provider once on the top of app

File: src/App.js [hunk 30]
```diff
@@ -2,6 +2,20 @@ import { useState, useEffect } from "react";
 import "./App.css";
 import Form from "./components/Form";
 import TodoList from "./components/TodoList";
+import {
+  Typography,
+  AppBar,
+  Toolbar,
+  Box,
+  Tooltip
+} from "@mui/material";
```
Comment (SMysakowski) [inline src/App.js]: Import directly. In this way, you add whole components to the app bundle

**Review by SMysakowski — CHANGES_REQUESTED** [review]

File: src/styles/MyButton.js (hunk 5 above)
Comment (zbednarz) [reply src/styles/MyButton.js]: Button import has been corrected. Import styled I have additional questions.

**Review by SMysakowski — APPROVED** [review]

**Pull request merged**